### PR TITLE
DEV: Modernization/cleanup

### PR DESCRIPTION
**Pull request by CvX**
* Component template colocation
* Native classes
* glimmer plugin connector
* `this.` in templates
* `?.` operator
* Merged `withPluginApi` calls
* Updated deps

### CODE REVIEW

File: assets/javascripts/discourse/components/assigned-topic-list-item.hbs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        newDotText=this.newDotText
+      }}
+    {{~/if}}
+  </span>
+  <div class="link-bottom-line">
+    {{#unless this.hideCategory}}
+      {{#unless this.topic.isPinnedUncategorized}}
+        {{category-link this.topic.category}}
+      {{/unless}}
+    {{/unless}}
```
Comment (janzenisaac) [inline assets/javascripts/discourse/components/assigned-topic-list-item.hbs:38]:
```suggestion
    {{#unless (and this.hideCategory this.topic.isPinnedUncategorized)}}
        {{category-link this.topic.category}}
    {{/unless}}
```

File: assets/javascripts/discourse/components/assigned-topic-list-item.hbs
Comment (janzenisaac) [inline assets/javascripts/discourse/components/assigned-topic-list-item.hbs:1]: There is so much `raw` in this file 😅

File: assets/javascripts/discourse/components/assigned-topic-list-item.js [hunk 2]
```diff
@@ -1,7 +1,8 @@
 import TopicListItem from "discourse/components/topic-list-item";
 import { equal } from "@ember/object/computed";
 
-export default TopicListItem.extend({
-  classNames: ["assigned-topic-list-item"],
-  isPrivateMessage: equal("topic.archetype", "private_message"),
-});
+export default class AssignedTopicListItem extends TopicListItem {
+  classNames = ["assigned-topic-list-item"];
```
Comment (janzenisaac) [inline assets/javascripts/discourse/components/assigned-topic-list-item.js:5]: is this still necessary?

File: assets/javascripts/discourse/templates/modal/assign-user.hbs [hunk 3]
```diff
@@ -52,7 +52,7 @@
       </label>
       <Textarea
         id={{"assign-modal-note"}}
```
Comment (janzenisaac) [inline assets/javascripts/discourse/templates/modal/assign-user.hbs:54]:
```suggestion
        id="assign-modal-note"
```

File: assets/javascripts/discourse/templates/modal/assign-user.hbs [hunk 4]
```diff
@@ -37,9 +37,9 @@
         <label>{{i18n "discourse_assign.assign_modal.status_label"}}</label>
         <ComboBox
           @id="assign-status"
```
Comment (janzenisaac) [inline assets/javascripts/discourse/templates/modal/assign-user.hbs:39]:
```suggestion
          id="assign-status"
```

File: plugin.rb [hunk 5]
```diff
@@ -5,7 +5,6 @@
 # version: 1.0.1
 # authors: Sam Saffron
 # url: https://github.com/discourse/discourse-assign
-# transpile_js: true
```
Comment (janzenisaac) [inline plugin.rb:8]: Personal question, but why are we changing this?

File: assets/javascripts/discourse-assign/connectors/groups-interaction-custom-options/assignable-interaction-fields.hbs [hunk 6]
```diff
@@ -1,17 +1,20 @@
...
+  <label class="control-label">
+    {{i18n "discourse_assign.admin.groups.manage.interaction.assign"}}
+  </label>
+
+  <label for="visiblity">
+    {{i18n
       "discourse_assign.admin.groups.manage.interaction.assignable_levels.title"
-    }}</label>
+    }}
+  </label>
```
[the remark this answers is not in the document]
Comment (CvX) [reply assets/javascripts/discourse-assign/connectors/groups-interaction-custom-options/assignable-interaction-fields.hbs:10]: I was trying to be conservative with the whitespace changes but decided to clean this one up and did eyeball it in the local env - looks good 😃

File: assets/javascripts/discourse/components/assigned-topic-list-item.hbs
Comment (CvX) [reply assets/javascripts/discourse/components/assigned-topic-list-item.hbs:1]: ✨ topic list! ✨ we'll be working on reworking all that soon enough 😁

File: plugin.rb (hunk 5 above)
Comment (CvX) [reply plugin.rb:8]: This option is no longer used - all plugin js is now transpiled. (that line was removed from the plugin skeleton a year ago 😃 https://github.com/discourse/discourse-plugin-skeleton/pull/21)

File: assets/javascripts/discourse/templates/modal/assign-user.hbs (hunk 4 above)
Comment (CvX) [reply assets/javascripts/discourse/templates/modal/assign-user.hbs:39]: (see the other select-kit comment)

File: assets/javascripts/discourse/components/assigned-topic-list-item.hbs (hunk 1 above)
Comment (CvX) [reply assets/javascripts/discourse/components/assigned-topic-list-item.hbs]: hah, in the end linter did force me to change it to `if` 😛

**Review by janzenisaac — APPROVED** [review]

**Pull request merged**